terraform/common/groups.tf

```diff
@@ -0,0 +1,182 @@
+}
+
+data aws_iam_policy_document allow_assume_role_billingmanager {
+  statement {
+    actions   = ["sts:AssumeRole"]
+    resources = ["arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/${aws_iam_role.billingmanager.name}"]
+  }
+}
+
+resource aws_iam_policy allow_assume_role_billingmanager {
+  name        = "allow_assume_role_billingmanager"
+  description = "Allow assuming billingmanager role"
+  policy      = data.aws_iam_policy_document.allow_assume_role_billingmanager.json
+}
+
+resource aws_iam_role_policy_attachment billingmanager_role_policies {
+  for_each = toset([
+    "arn:aws:iam::aws:policy/job-function/Billing",
+    "arn:aws:iam::aws:policy/ReadOnlyAccess"
+  ])
+
+  role       = aws_iam_role.billingmanager.name
+  policy_arn = each.value
+}
+
+
+# ReadOnly
+resource aws_iam_role readonly {
+  name               = "ReadOnly"
+  assume_role_policy = data.aws_iam_policy_document.assume_role_if_mfa_present.json
+}
+
+data aws_iam_policy_document allow_assume_role_readonly {
+  statement {
+    actions   = ["sts:AssumeRole"]
+    resources = ["arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/${aws_iam_role.readonly.name}"]
+  }
+}
+
+resource aws_iam_policy allow_assume_role_readonly {
+  name        = "allow_assume_role_readonly"
+  description = "Allow assuming readonly role"
+  policy      = data.aws_iam_policy_document.allow_assume_role_readonly.json
+}
+
+resource aws_iam_role_policy_attachment readonly_role_policies {
+  for_each = toset([
+    "arn:aws:iam::aws:policy/AmazonRoute53ReadOnlyAccess",
+    "arn:aws:iam::aws:policy/AmazonS3ReadOnlyAccess",
+    "arn:aws:iam::aws:policy/AWSCertificateManagerReadOnly",
+    "arn:aws:iam::aws:policy/CloudFrontReadOnlyAccess"
+  ])
+
+  role       = aws_iam_role.readonly.name
+  policy_arn = each.value
+}
+
+# SecretEditor
+resource aws_iam_role secreteditor {
+  name               = "SecretEditor"
+  assume_role_policy = data.aws_iam_policy_document.assume_role_if_mfa_present.json
+}
+
+data aws_iam_policy_document allow_assume_role_secreteditor {
+  statement {
+    actions   = ["sts:AssumeRole"]
+    resources = ["arn:aws:iam::${data.aws_caller_identity.current.account_id}:role/${aws_iam_role.secreteditor.name}"]
+  }
+}
+
+resource aws_iam_policy allow_assume_role_secreteditor {
+  name        = "allow_assume_role_secreteditor"
+  description = "Allow assuming secreteditor role"
+  policy      = data.aws_iam_policy_document.allow_assume_role_secreteditor.json
+}
+
+data aws_iam_policy_document parameter_store {
+  statement {
+    actions = [
+      "ssm:DescribeParameters",
+      "ssm:PutParameter",
+      "ssm:DeleteParameter",
+      "ssm:GetParameterHistory",
+      "ssm:GetParametersByPath",
+      "ssm:GetParameters",
+      "ssm:GetParameter",
+      "ssm:DeleteParameters"
+    ]
+    resources = [
+      "arn:aws:ssm:${var.region}:${data.aws_caller_identity.current.account_id}:parameter/${local.service_name}/dev/*",
```

Comment:
Why not `"arn:aws:ssm:${var.region}:${data.aws_caller_identity.current.account_id}:parameter/${local.service_name}/*"` ?